presto-server-rpm/src/main/resources/scripts/preinstall

```diff
@@ -0,0 +1,66 @@
+# Pre installation script
+
+# Ensure that the proper version of Java exists on the system
+
+java_version() {
+# The one argument is the location of java (either $JAVA_HOME or a potential
+# candidate for JAVA_HOME.
+  "$1"/bin/java -version 2>&1 | grep "java version" | awk '{ print substr($3, 2, length($3)-2); }'
+}
+
+java_vendor() {
+# The one argument is the location of java (either $JAVA_HOME or a potential
+# candidate for JAVA_HOME. 4
+# Returns the java vendor name. eg: Oracle Corporation
+  "$1"/bin/java -XshowSettings:properties -version 2>&1 | grep "java.vendor =" | awk '{ print $3 " " $4; }'
+}
+
+check_if_correct_java_version() {
+# The one argument is the location of java (either $JAVA_HOME or a potential
+# candidate for JAVA_HOME.
+  JAVA_VERSION=$(java_version "$1")
+  JAVA_VENDOR=$(java_vendor "$1")
+  if [[ ("$JAVA_VERSION" > "1.8") && ("$JAVA_VENDOR" = "Oracle Corporation") ]]; then
```

Comment:
We recommend running java 8u40 at a minimum, so you may want to test for the minor version, too.